tests/test_format_forecast.py

```diff
@@ -0,0 +1,133 @@
+"""
+Test Suite for `format_to_forecast_sql` Function
+
+This script validates the functionality of the `format_to_forecast_sql` function.
+It ensures that the function correctly creates a single ForecastSQL object with multiple
+ForecastValue entries.
+
+### How to Run the Tests:
+
+Run the entire suite:
+    pytest tests/test_format_forecast.py
+
+Run a specific test:
+    pytest tests/test_format_forecast.py::test_format_to_forecast_sql_real
+
+For verbose output:
+    pytest tests/test_format_forecast.py -v
+
+Note: 
+This test assumes a local PostgreSQL database configured with:
+    - Username: `postgres`
+    - Password: `12345`
+    - Database: `testdb`
+
+This setup is temporary for local testing and may require adjustment for CI/CD environments.
+"""
+
+import pytest
+from typing import Generator
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from nowcasting_datamodel.models.base import Base_Forecast
+from nowcasting_datamodel.models import MLModelSQL, ForecastSQL, ForecastValue
+from neso_solar_consumer.fetch_data import fetch_data
+from neso_solar_consumer.format_forecast import format_to_forecast_sql
+
+# Test configuration constants
+RESOURCE_ID = "db6c038f-98af-4570-ab60-24d71ebd0ae5"
+LIMIT = 5
+COLUMNS = ["DATE_GMT", "TIME_GMT", "EMBEDDED_SOLAR_FORECAST"]
+RENAME_COLUMNS = {
+    "DATE_GMT": "start_utc",
+    "TIME_GMT": "end_utc",
+    "EMBEDDED_SOLAR_FORECAST": "solar_forecast_kw",
+}
+MODEL_NAME = "real_data_model"
+MODEL_VERSION = "1.0"
+
+
+@pytest.fixture
+def db_session() -> Generator:
+    """
+    Fixture to set up and tear down a PostgreSQL database session for testing.
+
+    Database connection details (modify if required):
+        - Host: localhost
+        - User: postgres
+        - Password: 12345
+        - Database: testdb
+
+    Creates fresh tables before each test and cleans up after execution.
+    """
+    engine = create_engine("postgresql://postgres:12345@localhost/testdb")
```

Comment:
this looks good, you can do something like this ` with PostgresContainer("postgres:15.5") as postgres:` that just spins it up automatically

see https://github.com/openclimatefix/pv-site-datamodel/blob/main/tests/conftest.py#L26C4-L26C57

Btw, this is hard, so dont worry if this takes some time